cmd/get_logs.go

```diff
@@ -347,6 +380,28 @@ func (glc *getLogsCmd) downloadLogs(node v1.Node, client *ssh.Client) (string, e
 	return "", nil
 }
 
+func (glc *getLogsCmd) validateLogScript() error {
+	if glc.linuxScriptPath == "" && !glc.cs.Properties.MasterProfile.IsVHDDistro() {
+		if glc.controlPlaneOnly {
+			return errors.Errorf("No log collection script found for control plane nodes")
+		} else {
+			log.Warnf("Skipping master nodes nodes as no log collection script found")
+			glc.masterNodes = nil
+		}
+	}
+	for _, profile := range glc.cs.Properties.AgentPoolProfiles {
+		if glc.linuxScriptPath == "" && profile.OSType == "Linux" && !profile.IsVHDDistro() {
+			log.Warnf("Skipping nodes in linux agentpool %s as no log collection script found", profile.Name)
+			glc.linuxNodes = updateLinuxNodesList(glc.linuxNodes, profile.Name)
+		}
+		if glc.windowsScriptPath == "" && profile.OSType == "Windows" && !(glc.cs.Properties.WindowsProfile.WindowsPublisher == "microsoft-aks" && glc.cs.Properties.WindowsProfile.WindowsOffer == "aks-windows") {
+			log.Warnf("Skipping nodes in windows agentpools as no log collection script found")
```

Comment:
I am not sure this message will make sense to people without as much context as you. I would explain why this happens.

`Skipping windows nodes as flag '--windows-script' is not set and the pool's distro is not {name}`

Same for linux and master nodes

Reply:
Updated